ggshield/cmd/sca/scan.py

```diff
@@ -53,16 +61,25 @@ def scan_group(*args, **kwargs: Any) -> None:
 )
 @click.pass_context
 @display_sca_beta_warning
-def scan_diff_cmd(
+def scan_pre_commit_cmd(
     ctx: click.Context,
     directory: Optional[Path],
     ref: str,
     **kwargs: Any,
-) -> int:
+) -> SCAScanDiffOutput:
     """
-    Find SCA vulnerabilities in a git working directory, compared to git history.
+    Find SCA vulnerabilities in a git working directory, compared to HEAD.
     """
-    return 0
+    if directory is None:
+        directory = Path().resolve()
+
+    # Adds client and required parameters to the context
+    update_context(ctx)
+
+    result = sca_scan_diff(ctx, directory, "HEAD", True)
```

Comment:
nit: can you use named arguments here ? I'm not fan of having a `True` argument without a name, for readability.